src/actions/slider.js

```diff
@@ -0,0 +1,55 @@
+export function asSlider(node) {
+  let x;
+
+  function handleMousedown(e) {
+    x = typeof e.clientX == 'number' ? e.clientX : e.touches[0].clientX;
+    node.dispatchEvent(
+      new CustomEvent('slideStart', {
+        detail: { x }
+      })
+    );
+
+    window.addEventListener('mousemove', handleMousemove);
```

Comment:
Why window and not document.body?

Reply:
Actually examples that I looked used window) I never thought about difference.

Reply:
As I see, the only difference in my case is that document.body is closer to the source

Reply:
That's my point.
Using global object is tricky.
Imagin you will have to add another slider effect in the app. Where would you place the handler?